File: src/main/java/me/miki/shindo/management/cosmetic/cape/CapeCategory.java
package me.miki.shindo.management.cosmetic.cape;

import lombok.Getter;
import me.miki.shindo.management.language.TranslateText;
import me.miki.shindo.utils.animation.ColorAnimation;
import me.miki.shindo.utils.animation.simple.SimpleAnimation;

@Getter
public enum CapeCategory {
    ALL(TranslateText.ALL.getText()), MINECON("Minecon"), FLAG("Flags"), CARTOON("Cartoon"), CUSTOM("Custom");

    private final String name;
    private final SimpleAnimation backgroundAnimation = new SimpleAnimation();
    private final ColorAnimation textColorAnimation = new ColorAnimation();

    CapeCategory(String name) {
        this.name = name;
    }

}

File: src/main/kotlin/me/miki/shindo/management/cosmetic/cape/CapeCategory.kt
package me.miki.shindo.management.cosmetic.cape

import me.miki.shindo.management.language.TranslateText
import me.miki.shindo.ui.animation.v2.value.ColorAnimation
import me.miki.shindo.ui.animation.v2.value.SimpleAnimation

enum class CapeCategory(
    private val names: String,
) {
    ALL(TranslateText.ALL.getText()),
    MINECON("Minecon"),
    FLAG("Flags"),
    CARTOON("Cartoon"),
    CUSTOM("Custom"),
    ;

    private val backgroundAnimation = SimpleAnimation()
    private val textColorAnimation = ColorAnimation()

    fun getName(): String = names

    fun getBackgroundAnimation(): SimpleAnimation = backgroundAnimation

    fun getTextColorAnimation(): ColorAnimation = textColorAnimation
}
